RecFCCeeCalorimeter/src/components/CreateFCCeeCaloNeighbours.cpp

```diff
@@ -0,0 +1,488 @@
+    }
+    // make sure the ordering above is as in property activeFieldNamesNested
+    std::map<std::string, uint> activeVolumesNumbersNested;
+    for (const auto& name : m_activeFieldNamesNested) {
+      for (const auto& number : numVolumesMap) {
+        if (name == number.first) {
+          activeVolumesNumbersNested.insert(std::make_pair(number.first, number.second));
+        }
+      }
+    }
+
+    // Loop over all cells in the calorimeter and retrieve existing cellIDs
+    // Loop over active layers
+    std::vector<std::pair<int, int>> extrema;
+    extrema.push_back(std::make_pair(0, activeVolumesNumbersNested.find(m_activeFieldNamesNested[0])->second - 1));
+    extrema.push_back(std::make_pair(0, activeVolumesNumbersNested.find(m_activeFieldNamesNested[1])->second - 1));
+    extrema.push_back(std::make_pair(0, activeVolumesNumbersNested.find(m_activeFieldNamesNested[2])->second - 1));
+    // for layer 0 of HCal barrel
+    if (m_fieldNameNested == "system" && m_fieldValuesNested[iSys] == 8) {
+      extremaHCalFirstLayerPhi =
+          std::make_pair(0, activeVolumesNumbersNested.find(m_activeFieldNamesNested[1])->second - 1);
+      extremaHCalFirstLayerZ =
+          std::make_pair(0, activeVolumesNumbersNested.find(m_activeFieldNamesNested[2])->second - 1);
+    }
+    for (unsigned int ilayer = 0; ilayer < activeVolumesNumbersNested.find(m_activeFieldNamesNested[0])->second;
+         ilayer++) {
+      for (unsigned int iphi = 0; iphi < activeVolumesNumbersNested.find(m_activeFieldNamesNested[1])->second; iphi++) {
+        for (unsigned int iz = 0; iz < activeVolumesNumbersNested.find(m_activeFieldNamesNested[2])->second; iz++) {
+
+	  dd4hep::DDSegmentation::CellID cID = volumeId;
+          decoder->set(cID, m_activeFieldNamesNested[0], ilayer);
+	  decoder->set(cID, m_activeFieldNamesNested[1], iphi);
+	  decoder->set(cID, m_activeFieldNamesNested[2], iz);
+          
+          map.insert(std::pair<uint64_t, std::vector<uint64_t>>(
+              cID, det::utils::neighbours(*decoder, {m_activeFieldNamesNested[0], m_activeFieldNamesNested[1],
+                                                          m_activeFieldNamesNested[2]},
+                                               extrema, cID, {false, true, false}, true)));
+        }
+      }
+    }
+    if (msgLevel() <= MSG::DEBUG) {
+      std::vector<int> counter;
+      counter.assign(40, 0);
+      for (const auto& item : map) {
+        counter[item.second.size()]++;
+      }
+      for (uint iCount = 0; iCount < counter.size(); iCount++) {
+        if (counter[iCount] != 0) {
+          info() << counter[iCount] << " cells have " << iCount << " neighbours" << endmsg;
+        }
+      }
+    }
+  }
+
+  //////////////////////////////////////////////////
+  ///      BARREL: connection ECAL + HCAL        ///
+  /////////////////////////////////////////////////
+  int count=0;
+/*
```

Comment:
Is this peace of code intended to be used for ECAL-HCAL matching?

Reply:
this part needs to be implemented. I copied here the code that was pre-existing but I am gonna refactor it in the coming days, as soon as I find out the settings to use for the HCAL clustering.

Reply:
OK